### DISCUSSION TITLE
[Multiapp][Transfer]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#21536] [Q&A General] [answered]
Dear Users,

I am trying to make the following multi-app:

![image](https://user-images.githubusercontent.com/35205601/177950315-212b229e-23dc-468d-923f-a02c522b53ce.png)

I am trying to use _FullSolveMultiApp_ and _MultiAppCopyTransfer_, however, it doesn't work very well.

When I was trying, Subapp doesn't update value B from the main app after timestep 1.

The following lines are the multiapps and transfer I used.

Did I make any mistake here?

-----------------------------------------------------------------------
Code
```
[MultiApps]
  [sub_app]
    type = FullSolveMultiApp
    app_type = corrosionApp
    input_files = 'FakeTest2.i'
    clone_master_mesh = true # Use mesh of master.i in sub.i file
  []
[]
[Transfers]
  [pull]
    type = MultiAppCopyTransfer
    from_multi_app = sub_app
    variable = 'B'
    source_variable = 'B'
    execute_on = 'TIMESTEP_BEGIN'
  []
  [push]
    type = MultiAppCopyTransfer
    to_multi_app = sub_app
    source_variable = 'A'
    variable = 'A'
    execute_on = 'INITIAL TIMESTEP_END'
  []
[]
```
-----------------------------------------------------------------------

Best regards,
nakkyu

**Comment by GiudGiud** [comment, marked as answer]
I think you may want to execute all the transfers and the multiapp on timestep_end.

The ordering goes like this then:

main app executes
transfer to subapp of B
subapp executes
transfer to main app of A

then repeats if you have fixed point coupling iterations turned on

**Reply by coskrrb2002** [reply]
Thank you for your advice!

But I found I have an issue with the Terminator object.
